### PR TITLE
End support for python 3.4

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:* https://github.com/aws-cloudformation/cfn-python-lint/issues/1291

*Description of changes:*
- End support for python 3.4

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -42,21 +42,18 @@ def get_version(filename):
     packages=find_packages('src'),
     zip_safe=False,
     install_requires=[
-        'pyyaml<=5.2;python_version=="3.4"',
-        'pyyaml;python_version!="3.4"',
+        'pyyaml<=5.2',
```
Comment (PatMyron) [inline setup.py:45]:
```suggestion
        'pyyaml',
```

Comment (kddejong) [reply setup.py]: Fixed.

File: setup.py [hunk 2]
```diff
@@ -42,21 +42,19 @@ def get_version(filename):
...
+        'jsonpatch',
         'jsonschema~=3.0',
-        'pathlib2>=2.3.0;python_version<="3.4"',
-        'importlib_resources~=1.0.2;python_version=="3.4"',
-        'importlib_resources~=1.4;python_version<"3.7" and python_version!="3.4"',
+        'pathlib2>=2.3.0;python_version<"3.5"',
+        'importlib_resources~=1.4;python_version<"3.7"',
         'networkx~=2.4;python_version>="3.5"',
         'networkx<=2.2;python_version<"3.5"',
+        'pyrsistent<=0.16.0;python_version<"3.5"',
```
Comment (PatMyron) [inline setup.py:54]: for [Python 2](https://stackoverflow.com/questions/63802032/cfn-lint-failing-while-validating-cloudformation-templates)

**Comment by PatMyron** [comment]
Some other places to update when we deprecate Python 3.4 as well:
```bash
$ git grep ' 3\.4' .
CONTRIBUTING.md:1. You will need Python 2 >= 2.7 or Python 3 >= 3.4. Verify which version you have by running `python --version`. 
CONTRIBUTING.md:    pip install -e . # or "pip3 install -e ." if you are using Python 3.4+
README.md:Python 2.7+ and 3.4+ are supported.
scripts/install:            "and greater or Python 3.4 and greater.\n"
```
https://github.com/aws-cloudformation/cfn-python-lint/blob/9bb2d457b9d633910ee8b8f5afb68ab301ee20f7/README.md#L31

https://github.com/aws-cloudformation/cfn-python-lint/blob/4eccd3faa7775c16521e2b5b827b7e8c1f9bffd4/scripts/install#L188

https://github.com/aws-cloudformation/cfn-python-lint/blob/87566923ecee1ce7f808e7b72bd4a68338f51bb1/CONTRIBUTING.md#L24-L34

**Comment by PatMyron** [comment]
[no response to my email seeking internal package Python minor version deprecation guidance](https://list-archive.amazon.com/messages/10562905)

but SAM reaching out due to 3.4 being unsupported on their end:
https://github.com/aws/serverless-application-model/issues/1255, https://github.com/aws/serverless-application-model/pull/968, https://github.com/aws/serverless-application-model/pull/1256

**Comment by atkinsonm** [comment]
> [no response to my email seeking internal package Python minor version deprecation guidance](https://list-archive.amazon.com/messages/10562905)
> 
> but SAM reaching out due to 3.4 being unsupported on their end:
> [aws/serverless-application-model#1255](https://github.com/aws/serverless-application-model/issues/1255), [aws/serverless-application-model#968](https://github.com/aws/serverless-application-model/pull/968), [aws/serverless-application-model#1256](https://github.com/aws/serverless-application-model/pull/1256)

I suggest asking in #cloudformation-interest on Slack. Anecdotally, it looks like many internal packages have or are migrating off of Python 3.4, so I suspect that dropping support in cfn-lint won't have much negative impact.

**Comment by kddejong** [comment]
@atkinsonm and @PatMyron I've changed this PR a little.  Instead of just turning off python 3.4 all together I left some of the code as is.  I think what we can say this is using Python 3.4 is at your own risk and will not be actively tested against.  Additionally i don't think going forward we write code to handle specific 3.4 use cases.  Our new testing with github actions does not include python 3.4 either.

**Comment by kddejong** [comment]
Our daily downloads for python 3.4 since early December has been less than 20.  I'm hoping this should be good enough to close out the python 3.4 support portion of this.

**Comment by PatMyron** [comment]
@kddejong sorry, deleting `master` branch automatically closed this and not sure how to re-open

**Pull request closed without merging**